tests/test_trace.py

```diff
@@ -130,8 +130,15 @@ def to_dict(self):
 
 
 def setup_function(func):
+    trace_factory.use_single_trace = True
```

Comment:
Previously it used to run with multiple traces (just because the order of the tests), and some of the fixes I did here were needed to make it run like that - which is the default way